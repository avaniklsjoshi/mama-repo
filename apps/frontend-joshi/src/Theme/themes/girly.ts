export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
