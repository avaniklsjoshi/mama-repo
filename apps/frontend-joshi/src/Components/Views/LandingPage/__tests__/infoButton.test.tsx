/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import girly from "../../../../Theme/themes/girly";
import InfoButton from "../Timeline/InfoButton";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: any) => key, i18n: (key: any) => key })
}));

describe("Landing Page", () => {
  it("renders InfoButton without data", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <InfoButton
          projects={""}
          content={{
            tech: [],
            role: []
          }}
        />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.queryByText("timelineInfoTitle")).not.toBeInTheDocument();
    expect(screen.queryByText("timelineInfoRole")).not.toBeInTheDocument();
  });
  it("renders InfoButton with data", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <InfoButton
          projects={"CMS"}
          content={{
            tech: ["Javascript"],
            role: ["Software Developer"]
          }}
        />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("Software Developer")).toBeInTheDocument();
    expect(screen.getByText("Javascript")).toBeInTheDocument();
  });
});
