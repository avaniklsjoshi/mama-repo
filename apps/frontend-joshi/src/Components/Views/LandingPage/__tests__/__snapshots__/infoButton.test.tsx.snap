// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders InfoButton with data 1`] = `
<div>
  <div
    class="info-container"
  >
    <div
      class="tech-container"
    >
      <div
        class="info-title"
      >
        timelineInfoTitle
         
        timelineInfoTitleSingle
        -
      </div>
      <ul>
        <li>
          Javascript
        </li>
      </ul>
    </div>
    <div
      class="role-container"
    >
      <div
        class="info-title"
      >
        timelineInfoRole
         
        timelineInfoTitleSingle
        -
      </div>
      <ul>
        <li>
          Software Developer
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Landing Page renders InfoButton without data 1`] = `
<div>
  <div
    class="info-container"
  >
    <div
      class="tech-container"
    >
      <div
        class="info-title"
      >
        timelineInfoTitle
         
        timelineInfoTitleSingle
        -
      </div>
      <ul />
    </div>
    <div
      class="role-container"
    >
      <div
        class="info-title"
      >
        timelineInfoRole
         
        timelineInfoTitleSingle
        -
      </div>
      <ul />
    </div>
  </div>
</div>
`;
